admin-dev/themes/new-theme/js/pages/product/components/combination-list/bulk-delete-handler.ts

```diff
@@ -61,46 +61,42 @@ export default class BulkDeleteHandler {
-
-    if (!(bulkDeleteBtn instanceof HTMLButtonElement)) {
-      console.error(`${CombinationMap.bulkDeleteBtn} must be a HTMLButtonElement`);
-
-      return;
-    }
-
-    bulkDeleteBtn.addEventListener('click', async () => {
-      const selectedCombinationIds = await this.bulkChoicesSelector.getSelectedIds();
-
-      try {
-        const selectedCombinationsCount = selectedCombinationIds.length;
-        const confirmLabel = bulkDeleteBtn.dataset.modalConfirmLabel
-          ?.replace(/%combinations_number%/, String(selectedCombinationsCount));
-
-        const modal = new ConfirmModal(
-          {
-            id: 'modal-confirm-delete-combinations',
-            confirmTitle: bulkDeleteBtn.innerHTML,
-            confirmMessage: bulkDeleteBtn.dataset.modalMessage,
-            confirmButtonLabel: confirmLabel,
-            closeButtonLabel: bulkDeleteBtn.dataset.modalCancelLabel,
-            closable: true,
-          },
-          async () => {
-            await this.bulkDelete(selectedCombinationIds);
-          },
-        );
-        modal.show();
-      } catch (error: any) {
-        const errorMessage = error.response?.JSON ?? error;
-        $.growl.error({message: errorMessage});
-      }
+    const bulkDeleteButtons = document.querySelectorAll<HTMLButtonElement>(CombinationMap.bulkDeleteBtn);
+
+    bulkDeleteButtons.forEach((bulkDeleteBtn: HTMLButtonElement) => {
+      bulkDeleteBtn.addEventListener('click', async () => {
+        const selectedCombinationIds = await this.bulkChoicesSelector.getSelectedIds();
+
+        try {
+          const selectedCombinationsCount = selectedCombinationIds.length;
+          const confirmLabel = bulkDeleteBtn.dataset.modalConfirmLabel
+            ?.replace(/%combinations_number%/, String(selectedCombinationsCount));
+
+          const modal = new ConfirmModal(
+            {
+              id: 'modal-confirm-delete-combinations',
+              confirmTitle: bulkDeleteBtn.innerHTML,
+              confirmMessage: bulkDeleteBtn.dataset.modalMessage,
+              confirmButtonLabel: confirmLabel,
+              closeButtonLabel: bulkDeleteBtn.dataset.modalCancelLabel,
+              closable: true,
+            },
+            async () => {
+              await this.bulkDelete(selectedCombinationIds, bulkDeleteBtn);
+            },
+          );
+          modal.show();
+        } catch (error: any) {
+          const errorMessage = error.response?.JSON ?? error;
+          $.growl.error({message: errorMessage});
+        }
+      });
     });
   }
 
-  private async bulkDelete(combinationIds: number[]): Promise<void> {
-    const $bulkDeleteBtn = $(CombinationMap.bulkDeleteBtn);
-    const bulkChunkSize = Number($bulkDeleteBtn.data('bulkChunkSize'));
+  private async bulkDelete(combinationIds: number[], bulkDeleteBtn: HTMLButtonElement): Promise<void> {
+    // const $bulkDeleteBtn = $(bulkDeleteBtn);
```

Comment:
Good thing you removed the usage of JQuery, the I guess this line can be removed as well